### PR TITLE
Add creation of user on the fly JWT

### PR DESCRIPTION
**Pull request by dragonsahead**
This is something that comes up pretty often on convos

### CODE REVIEW

**Comment by nllho** [comment]
@paoliniluis Do you know if this note applies to accounts created with JWT auth:

> Note that Metabase accounts created with Google Sign-In do not have passwords and must use Google to sign in to Metabase.

From:
https://www.metabase.com/docs/latest/people-and-groups/google-and-ldap#creating-metabase-accounts-with-google-sign-in

File: docs/people-and-groups/authenticating-with-jwt.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ redirect_from:
 
 {% include plans-blockquote.html feature="JWT-based authentication" %}
 
-You can connect Metabase to your identity provider using JSON Web Tokens (JWT) to authenticate people.
+You can connect Metabase to your identity provider using JSON Web Tokens (JWT) to authenticate people. If the user doesn't exist, Metabase will create it on the fly.
```
Comment (nllho) [inline docs/people-and-groups/authenticating-with-jwt.md:11]:
```suggestion
You can connect Metabase to your identity provider using JSON Web Tokens (JWT) to authenticate people. If the user doesn't exist, Metabase will create an account on the fly.

Metabase accounts created with an IdP login don't have passwords. People who sign up for Metabase using an IdP must continue to use the IdP to log into Metabase.
```

We might want to copy this note over to the SAML doc as well, if it is a common question (and if the note about "no passwords" applies to all accounts created with SSO).

https://www.metabase.com/docs/latest/people-and-groups/authenticating-with-saml

**Review by nllho — APPROVED** [review]

**Comment by paoliniluis** [comment]
> @paoliniluis Do you know if this note applies to accounts created with JWT auth:
> 
> > Note that Metabase accounts created with Google Sign-In do not have passwords and must use Google to sign in to Metabase.
> 
> From: https://www.metabase.com/docs/latest/people-and-groups/google-and-ldap#creating-metabase-accounts-with-google-sign-in

yup! it's the same in this situation

Regarding your SAML note, it's exactly the same

**Comment by paoliniluis** [comment]
@nllho just added the same note to SAML. If this is fine, we should force merge

**Review by ranquild — APPROVED** [review]

**Pull request merged**